### PR TITLE
Allow spin_up to be run on CI

### PR DESCRIPTION
**Pull request by swebb**
Now you can run it on CI

### CODE REVIEW

File: lib/tasks/spin_up.rake [hunk 1]
```diff
@@ -3,28 +3,38 @@ require 'yaml'
...
 
 class SpinUpHelper
+  CONFIGS = [
+    "config/secrets.yml",
+    "config/database.yml",
+    "config/omniauth.yml",
+    "config/redis.yml",
+    "config/healthway.yml",
+    "config/ldap.yml"
+  ]
```
Comment (jordanmaguire) [inline lib/tasks/spin_up.rake]: many of these are wrong :)

File: lib/tasks/spin_up.rake [hunk 2]
```diff
@@ -3,28 +3,38 @@ require 'yaml'
 
 desc "Setup config files for database, and secrets. Setup database"
 task :spin_up do
-  configs = ["config/secrets.yml", "config/database.yml"]
+  SpinUpHelper.new(:test).spin_up
```
Comment (jordanmaguire) [inline lib/tasks/spin_up.rake]: This should be :dev I think

File: lib/tasks/spin_up.rake [hunk 3]
```diff
@@ -32,28 +42,26 @@ class SpinUpHelper
...
   end
 
   def setup_database
-    command = "rake db:create db:migrate && rake db:seed"
+    if @env == :development
+      command = "rake db:create db:migrate && rake db:seed"
+    elsif @env == :test
+      command = "RAILS_ENV=test rake db:drop db:create db:migrate"
+    else
+      raise "unknown environment"
```
Comment (jordanmaguire) [inline lib/tasks/spin_up.rake]: :+1:

**Comment by jordanmaguire** [comment]
:+1:

File: lib/tasks/spin_up.rake (hunk 2 above)
Comment (Ruxton) [inline lib/tasks/spin_up.rake]: shouldn't it just be Rails.env ?

File: lib/tasks/spin_up.rake [hunk 4]
```diff
@@ -32,28 +42,26 @@ class SpinUpHelper
...
+    yaml["test"]["secret_key_base"] = SecureRandom.hex(64)
     write_yaml(yaml, filename)
   end
 
   def setup_database
-    command = "rake db:create db:migrate && rake db:seed"
+    if @env == :development
+      command = "rake db:create db:migrate && rake db:seed"
+    elsif @env == :test
+      command = "RAILS_ENV=test rake db:drop db:create db:migrate"
```
Comment (Ruxton) [inline lib/tasks/spin_up.rake]: development db:migrate now prepares the test db, so this is kinda pointless

File: lib/tasks/spin_up.rake (hunk 3 above)
Comment (Ruxton) [inline lib/tasks/spin_up.rake]: Would like this to have basic DB spin as the fall back, I have to spin up up staging and production.

File: lib/tasks/spin_up.rake (hunk 4 above)
Comment (jordanmaguire) [inline lib/tasks/spin_up.rake]: is for CI

File: lib/tasks/spin_up.rake [hunk 5]
```diff
@@ -32,28 +42,26 @@ class SpinUpHelper
   def setup_secrets_config
     filename = "config/secrets.yml"
     yaml = YAML.load_file(filename)
-    yaml["development"]["secret_key_base"] = SecureRandom.hex(128)
-    yaml["test"]["secret_key_base"] = SecureRandom.hex(128)
+    yaml["development"]["secret_key_base"] = SecureRandom.hex(64)
+    yaml["test"]["secret_key_base"] = SecureRandom.hex(64)
```
Comment (Ruxton) [inline lib/tasks/spin_up.rake]: Why did these go down in size??

**Comment by lordmortis** [comment]
Another question: We should really be moving this into the bin/setup file because it looks to me like that's the advocated default for all ruby projects? (or at least the parts of this that can be done without spooling up rails)

Also, you can't bundle or setup databases inside a rake task *unless* they are already made AFAIK?

**Comment by jordanmaguire** [comment]
> Also, you can't bundle or setup databases inside a rake task unless they are already made AFAIK?

You can't bundle - bundling would mean that rake isn't installed so it'd just shit its britches.

You can setup DBs, tho.

**Comment by jordanmaguire** [comment]
> Another question: We should really be moving this into the bin/setup file because it looks to me like that's the advocated default for all ruby projects? (or at least the parts of this that can be done without spooling up rails)

Could you show me some docs on this?

I'm happy to move it up in tharr once I understand what that file is

**Comment by jordanmaguire** [comment]
https://robots.thoughtbot.com/bin-setup

**Comment by jordanmaguire** [comment]
Let's go with the spin_up file.

File: lib/tasks/spin_up.rake (hunk 5 above)
Comment (swebb) [inline lib/tasks/spin_up.rake]: I was using SecureRandom.hex wrong. It outputs twice the number you put in, so 64 will return 128 characters.

**Comment by swebb** [comment]
Closing this and will re-open a new PR with bin/setup

**Comment by pzi** [comment]
Can this branch be tossed?

**Pull request closed without merging**